Toast feat

nhattlm3173 opened the pull request.

nhattlm3173 committed:
add config prettier, husky, lint-staged

nhattlm3173 committed:
test husky

nhattlm3173 committed:
test husky

nhattlm3173 committed:
test husky lint staged with tsx

nhattlm3173 committed:
test husky lint staged with tsx

nhattlm3173 committed:
Adds ESLint cache file.

Adds the .eslintcache file to improve ESLint performance by caching linting results.

nhattlm3173 committed:
Adds basic structure for new pages

Sets up initial files and components for the blog, docs, and showcase pages.

Also, this commit refactors the layout component to improve code readability.

nhattlm3173 committed:
Updates eslint cache

Refreshes the eslint cache to include changes in new files.

This ensures that eslint correctly analyzes and reports issues in the updated codebase.

nhattlm3173 committed:
feat: Implements a toast notification system

nhattlm3173 committed:
Updates eslint cache

Updates the .eslintcache file to reflect changes in the project's linting status and include new files. This ensures that eslint correctly identifies and reports any linting issues in the codebase.

MarsCodeJS commented on src/component/toast/toast-context.tsx:
```diff
@@ -0,0 +1,33 @@
+  );
+};
+
+export const useToast = () => {
+  const context = useContext(ToastContext);
+  if (!context) {
+    throw new Error("useToast must be used within a ToastProvider");
+  }
+  return context;
+};
```
Add an empty line between conditional statements and variable declarations or return statements to improve readability.

For example:
```ts
export const useToast = () => {
  const context = useContext(ToastContext);

  if (!context) {
    throw new Error('useToast must be used within a ToastProvider');
  }

  return context;
};
```

MarsCodeJS commented on .prettierrc:
```diff
@@ -0,0 +1,10 @@
+{
+  
+    "plugins": ["prettier-plugin-tailwindcss"],
+    "semi": true,
+    "tabWidth": 2,
+    "trailingComma": "all",
+    "endOfLine": "auto",
+    "printWidth": 100
+  }
+  
```
Use standard Prettier formatting in the `.prettierrc` configuration file.

For example:
```
{
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "es5",
  "printWidth": 100,
  "bracketSpacing": true,
  "arrowParens": "avoid",
  "endOfLine": "lf",
  "bracketSameLine": false
}

```

MarsCodeJS commented on src/app/layout.tsx:
```diff
@@ -24,10 +26,11 @@ export default function RootLayout({
-      >
-        {children}
+      <body className={`${geistSans.variable} ${geistMono.variable} antialiased`}>
+        <ToastProvider>
+          {children}
+          <ToastContainer />
+        </ToastProvider>
```
Make sure to separate providers

For example:
`app/Providers.tsx`
```ts
import { ToastProvider } from '@/component/Toast';

export default async function Providers({ children }: { children: React.ReactNode }) {
  return <ToastProvider>{children}</ToastProvider>;
}

```

`app/layout.tsx`
```ts
export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en" className="mdl-js">
      <body className={`${geistSans.variable} ${geistMono.variable} antialiased`}>
        <Providers>
          {children}
        </Providers>
      </body>
    </html>
  );
}
```

MarsCodeJS commented on src/component/toast/toast-context.tsx:
Component names must use CamelCase — do not use lowercase paths like `src/component/toast/toast-context.tsx`.

Must be `src/components/Toast/index.tsx`

MarsCodeJS commented on src/component/toast/toast-context.tsx:
Add an empty line between conditional/loop/return statements and the following code block for better readability

nhattlm3173 committed:
Refactors code for improved maintainability

Refactors code to enhance maintainability and readability.

- Updates Prettier configuration for consistent code formatting.
- Updates imports and removes unused components related to toast notifications, integrating toast functionality directly within the ToastProvider.
- Adds animations for toast notifications.

nhattlm3173 committed:
Updates .eslintcache

Updates the .eslintcache file to reflect recent code changes and ensure accurate linting results.
